tensorboard/webapp/plugins/plugins_container_test.ts

```diff
@@ -140,6 +139,28 @@ describe('plugins_component', () => {
     store.refreshState();
   }
 
+  /**
+   * Configures default behavior of the MockStore.
+   * Call this only after all providers have been configured as it will freeze
+   * the TestBed configuration.
```

Comment:
I find this refactor a bit unwieldy and I think it can be improved in two ways:
```ts
// 1. factor everything in beforeEach as `setup` method that does both TestBed configuration AND mocking the store with default values
function setup(providersOverride: someType[]|undefined) {
 await TestBed.configureTestingModule({
      providers: providersOverride ?? [
     ...
 });
 store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
  store.overrideSelector(getPlugins, PLUGINS);
  ...
}

// Method 2
describe('plugin_component with PluginApiModule', () > {
  // current test block... 
});

describe('plugin_container without PluginApiHostModule', () => {
  beforeEach(() => {
    // do not provide the PluginApiModule and see how entire component behave without it
    // currently, we are just testing one method so even if other places break, we would not know.
  });
});
```

WDYT?

Reply:
I did method 1 with some inspiration from method 2 as well.